contrib/babelfishpg_tsql/src/pltsql_ruleutils.c

```diff
@@ -377,6 +379,80 @@ tsql_get_constraintdef(PG_FUNCTION_ARGS)
 	PG_RETURN_TEXT_P(string_to_text(res));
 }
 
+PG_FUNCTION_INFO_V1(tsql_get_expr);
+
+/* ----------
+ * tsql_get_expr          - Decompile an expression tree
+ *
+ * Input: an expression tree in nodeToString form, and a relation OID
+ *
+ * Output: reverse-listed expression
+ *
+ * Currently, the expression can only refer to a single relation, namely
+ * the one specified by the second parameter.  This is sufficient for
+ * partial indexes, column default expressions, etc.  We also support
+ * Var-free expressions, for which the OID can be InvalidOid.
+ * ----------
+ */
+
+Datum
+tsql_get_expr(PG_FUNCTION_ARGS)
+{
+    text       *expr = PG_GETARG_TEXT_PP(0);
+    Oid         relid = PG_GETARG_OID(1);
+    int         prettyFlags;
+    char       *relname;
+
+    prettyFlags = PRETTYFLAG_INDENT;
+
+    if (OidIsValid(relid))
```

Comment:
we have various combination of relid being valid/invalid and relname being NULL/NOT NULL, do we have test cases to cover each scenario?

Reply:
Added such testcases